io_scene_nif/modules/nif_import/property/material/__init__.py

```diff
@@ -79,47 +99,29 @@ def set_material_vertex_mapping(self, b_mesh, f_map, n_uvco):
                 else:
                     # non-textured material: vertex colors influence color
                     b_mat.use_vertex_color_paint = True
-
-            # if there's a base texture assigned to this material display it in Blender's 3D view, but only if there are UV coordinates
-            if mbasetex and mbasetex.texture and n_uvco:
-                image = mbasetex.texture.image
-                if image:
-                    for b_polyimage_index in f_map:
-                        if b_polyimage_index is None:
-                            continue
-                        tface = b_mesh.uv_textures.active.data[b_polyimage_index]
-                        tface.image = image
     """
 
     @staticmethod
     def import_material_specular(b_mat, n_specular_color):
-        b_mat.specular_color.r = n_specular_color.r
-        b_mat.specular_color.g = n_specular_color.g
-        b_mat.specular_color.b = n_specular_color.b
+        b_mat.specular_color = (n_specular_color.r, n_specular_color.g, n_specular_color.b)
 
     @staticmethod
     def import_material_emissive(b_mat, n_emissive_color):
-        b_mat.niftools.emissive_color.r = n_emissive_color.r
-        b_mat.niftools.emissive_color.g = n_emissive_color.g
-        b_mat.niftools.emissive_color.b = n_emissive_color.b
+        b_mat.niftools.emissive_color = (n_emissive_color.r, n_emissive_color.g, n_emissive_color.b)
 
     @staticmethod
     def import_material_diffuse(b_mat, n_diffuse_color):
-        b_mat.diffuse_color.r = n_diffuse_color.r
-        b_mat.diffuse_color.g = n_diffuse_color.g
-        b_mat.diffuse_color.b = n_diffuse_color.b
-        b_mat.diffuse_intensity = 1.0
+        b_mat.diffuse_color = (n_diffuse_color.r, n_diffuse_color.g, n_diffuse_color.b, 1.0)
+        # b_mat.diffuse_intensity = 1.0
```

Comment:
Would add a todo if we need to support otherwise can be removed if nolonger needed

Reply:
I think it no longer exists in the new API and was just there to ensure everything looked nice under the old API